noodles-editor/src/noodles/noodles.tsx

```diff
@@ -148,9 +149,20 @@ function useTheatreJs(projectName?: string) {
 
 // Not using the top-level sheet since a Noodles theatre sheet and project are dynamically created.
 // Also, the top-level sheet is used for theatre-managed project files, whereas a Noodles project file is managed within this visType.
+// Helper to extract projectId from current URL
+function getProjectIdFromUrl(): string | null {
```

Comment:
There's a nearly identical `getProjectId` in globals.ts. Maybe just put `const queryParams = new URLSearchParams(window.location.search)` inside and maintain one?